### PR TITLE
Fix `configPath` caching

### PR DESCRIPTION
**Pull request by kanongil**
See #7535 for details.

### CODE REVIEW

**Comment by rwjblue** [comment]
@kanongil - Mind rebasing? There was an issue with CI that was recently resolved and I believe this will be passing after a rebase...

**Comment by kanongil** [comment]
Rebased and pushed.

It seems my test fails on windows because the expect relies on unix path names:
```
expected 'C:\\projects\\ember-cli\\tests\\fixtures\\app-import\\config\\environment' to include 'app-import/config/environment'
```

File: tests/unit/broccoli/ember-app-test.js [hunk 1]
```diff
@@ -49,6 +49,21 @@ describe('EmberApp', function() {
...
+      };
+
+      expect(project.config('development')).to.deep.equal({});
+
+      new EmberApp({
+        project,
+        configPath: '../../app-import/config/environment',
+      });
+
+      expect(project.config('development').path).to.contain('app-import/config/environment');
```
Comment (rwjblue) [inline tests/unit/broccoli/ember-app-test.js]: Maybe use path.join or normalize for the expected value?

Comment (rwjblue) [reply tests/unit/broccoli/ember-app-test.js]: @kanongil - ping?  I believe fixing this will fix the current failure...

**Comment by Turbo87** [comment]
@kanongil friendly reminder

File: tests/unit/broccoli/ember-app-test.js (hunk 1 above)
Comment (locks) [reply tests/unit/broccoli/ember-app-test.js]: I rebased and addressed this comment.

**Pull request merged**